### DISCUSSION TITLE
Is there anyway to protect the derivations endpoint?

### DISCUSSION THREAD

**Discussion by gczh** [shrinerb/shrine#648] [Q&A] [answered]
From what I understand (i might be wrong!), the derivations endpoint allows one to easily generate variants on the fly. 

However, if the `upload: true` setting is turned on, would it be possible for someone to abuse the endpoint and generate infinite variations? With each variation, that would saturate the file storage provider (AWS S3 / Cloudflare R2). 

If so, then that would cost quite a bit for the app?

Is there any way then to protect the derivations endpoint from such situations?

**Comment by janko** [comment, marked as answer]
Yes, derivation URLs are signed, so only the app can generate them - https://shrinerb.com/docs/plugins/derivation_endpoint#how-it-works.
